app/factory/Layer.js

```diff
@@ -0,0 +1,184 @@
+/**
+ * [extend description]
+ * @type {String}
+ */
+Ext.define('CpsiMapview.factory.Layer', {
+    alternateClassName: 'LayerFactory',
+    requires: [],
+
+    singleton: true,
+
+    /**
+     * Creates an OpenLayer layer object from a JSON config
+     *
+     * @param  {Object} layerConf  The configuration object
+     * @return {ol.layer.Base} OL layer object
+     */
+    createLayer: function(layerConf) {
+
+        var layerType = layerConf.layerType;
+        var mapLayer;
+
+        switch (layerType) {
+        case 'wms':
+            mapLayer = LayerFactory.createWms(layerConf);
+            break;
+        case 'wfs':
+            mapLayer = LayerFactory.createWFS(layerConf);
+            break;
+        case 'esrirest':
+            mapLayer = LayerFactory.createEsriRest(layerConf);
+            break;
+        case 'osm':
+            mapLayer = LayerFactory.createOsm(layerConf);
+            break;
+        case 'empty':
+            mapLayer = LayerFactory.createEmptyLayer(layerConf);
+            break;
+        case 'bing_aerial':
+            mapLayer = LayerFactory.createBing(layerConf, 'Aerial');
+            break;
+        case 'google_roadmap':
+            mapLayer = LayerFactory.createGoogle(layerConf, 'google.maps.MapTypeId.ROADMAP');
+            break;
+        case 'google_terrain':
+            mapLayer = LayerFactory.createGoogle(layerConf, 'google.maps.MapTypeId.TERRAIN');
+            break;
+        case 'google_hybrid':
+            mapLayer = LayerFactory.createGoogle(layerConf, 'google.maps.MapTypeId.HYBRID');
+            break;
+        case 'google_satellite':
+            mapLayer = LayerFactory.createGoogle(layerConf, 'google.maps.MapTypeId.SATELLITE');
+            break;
```

Comment:
We either need a 3rd party plugin for these or remove them altogether. Lets create an issue

Reply:
See #8